### ISSUE TITLE
mlp multi layer

### ISSUE THREAD

**Issue by timcdlucas** [topepo/caret#279]
Hi,

I am hoping to use multi layer neural networks, with weight decay for classification. neuralnet is regression only, dnn has no weight decay.

So I think mlpWeightDecay is my best bet. However, I can't see a way to make this train multi layer networks through caret because the `size` argument in RSNNS::mlp takes a vector.

Is there a way of passing vectors to tuning parameters? Or do I need to rewrite the mlpWeightDecay caret model so that it takes separate tuning paramers (layer1, layer2, layer3 or something).

Thanks a lot in advance.

**Comment by fjeze** [comment]
Hi,

I had the same problem, I rewrote both mlp and mlpWeightDecay. 

You can use it by installing 
```
devtools::install_github('fjeze/caret/pkg/caret')
```

As tuning parameters it takes layer1, layer2, layer3 instead of size.

It was a quick fix so hopefully it is good enough.

cheers

**Comment by timcdlucas** [comment]
Thank you very much! I will do that and close this issue.

**Comment by zachmayer** [comment]
@fjeze Please feel free to make a pull request with your fix.  You can also add unit tests to ensure it doesn't break in the future.

**Comment by topepo** [comment]
@fjeze It would be a good idea to give the new protocols different names to preserve the old ones. 

Unit tests would be great, but add [Regression test files](https://github.com/topepo/caret/blob/master/RegressionTests/Code/mlpWeightDecay.R) too. Those should be almost copy/paste from the existing versions.

Thanks,

Max

**Comment by fjeze** [comment]
@zachmayer Where are the unit tests?

@topepo I will make it different classes. I have also have an alteration to the svmLinear2 class to add the gamma. Should I include that class as well?

**Comment by zachmayer** [comment]
https://github.com/topepo/caret/tree/master/pkg/caret/tests/testthat

Only tests starting with `test` will be run.

**Comment by topepo** [comment]
For that one, either way is fine with me.

**Comment by timcdlucas** [comment]
@fjeze Glad you're doing a PR. Your mlp models are good.

**Comment by fjeze** [comment]
i have done the pull see #287 

Cheers
